### DISCUSSION TITLE
Android Auto: open OSMAND+ with directions from another app

### DISCUSSION THREAD

**Discussion by Tiemichael** [osmandapp/OsmAnd#19524] [Q&A] [answered]
Is it possible to open OSMAND+ (version 4.7.6 ) with directions from another app in Android Auto?

I am trying to develop an Android Auto app, which is essentially a POI-database.
By clicking on the POI, Android Auto should open the last used Navigation App (Osmand+), and either show the Target Destination or start navigating to the Target Destination. (Which works fine without Android Auto).
All I see is Osmand+ opening, but there is no indication of receiving and coordinates.

My app is sending an intent (according to Android Auto Documentation) with "action = CarContext.ACTION_NAVIGATE", and "geo:Lat,Lon".

<
GH.NavClientManager     com...e.android.projection.gearhead  W  No Navigation Client Source for ComponentInfo{net.osmand.plus/net.osmand.plus.auto.NavigationCarAppService}
GH.NavClientManager     com...e.android.projection.gearhead  E  No corresponding nav client source for nav package: net.osmand.plus
CarApp.H.Tem            com...e.android.projection.gearhead  I  Intent received in Gearhead is Intent { act=android.intent.action.VIEW dat=https://maps.google.com/... flg=0x4000000 cmp=net.osmand.plus/.auto.NavigationCarAppService (has extras) } with Intent Data as https://maps.google.com/maps?nav=1&q=1.3548115,103.8775835 and Intent Action as android.intent.action.VIEW
CarApp.H.Tem            com...e.android.projection.gearhead  I  Gearhead Converted Intent is Intent { act=androidx.car.app.action.NAVIGATE dat=geo:1.3548115,103.8775835 flg=0x4000000 cmp=net.osmand.plus/.auto.NavigationCarAppService (has extras) } with Intent Data as geo:1.3548115,103.8775835 and Intent Action as androidx.car.app.action.NAVIGATE
CarApp.H.Tem            com...e.android.projection.gearhead  W  Binding to an already bound app with updatedAppKey as AppKey(componentName=ComponentInfo{net.osmand.plus/net.osmand.plus.auto.NavigationCarAppService}, profileType=PERSONAL) and current set of connected apps as [AppKey(componentName=ComponentInfo{net.osmand.plus/net.osmand.plus.auto.NavigationCarAppService}, profileType=PERSONAL)]
SurfaceRenderer         net.osmand.plus                      I  setupOffscreenRenderer
NavigationSession       net.osmand.plus                      I  In onNewIntent() Intent { act=androidx.car.app.action.NAVIGATE dat=0/main flg=0x4000000 cmp=net.osmand.plus/.auto.NavigationCarAppService (has extras) }
>
(If I select Google Maps as Navigation app, the target-destination is displayed, and I can click Navigate)

As my app is not yet published, I can only test it via the Android Auto HeadUnit Emulator.

This seems the same as [https://github.com/osmandapp/OsmAnd/issues/19130] - Issue which is reported as closed.
(OSMAND+ V4.7.6)

Thanks!

**Comment by Tiemichael** [comment, marked as answer]
Sorry,
but I realized the issue was related to the phone I used as Head-Unit Server running Android 9.
Using another phone running Android 14 shows NO problem.

**Reply by w201** [reply]
Just wonder why it had a problem on Android 9?
